File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<bool> flag) {
    vector<int> answer;
    int size = arr.size();

    for (int i = 0; i < size; i++) {
        if (flag[i]) {
            for (int j = 0; j < (arr[i] * 2); j++) {
                answer.push_back(arr[i]);
            }
        } else {
            for (int j = 0; j < arr[i]; j++) {
                answer.pop_back();
            }
        }
    }

    return answer;
}

File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.swift
import Foundation

func solution(_ arr:[Int], _ flag:[Bool]) -> [Int] {
    
    var result: [Int] = []
    
    for i in 0..<arr.count {
        flag[i] ? result.append(contentsOf: repeatElement(arr[i], count: arr[i]*2)) : result.removeLast(arr[i])
    }
    
    return result
}